### DISCUSSION TITLE
220523 유콩 인터뷰 피드백

### DISCUSSION THREAD

**Discussion by YJGwon** [woowacourse-study/2022-lv2-effective-java-interview#13] [General]
### 인터뷰어 -> 인터뷰이

> 점수를 매겨주셔도 좋고, 체크 형식으로 작성해주신 뒤 짧은 코멘트를 남겨주셔도 좋습니다. 해당 항목에 대해 자유롭게 의견을 남겨주세요.

- 말하기 태도, 어조가 적절했나요?
- 전달하고자 하는 바를 명료하게 전달했나요?
- 본인만의 주관이 잘 드러났나요?
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- 모르는 부분에 대해 잘 대처했나요?

### 인터뷰이 -> 인터뷰어

> 인터뷰어로써의 역량을 키우려는 것이 아닌 저희 스터디의 원활한 진행을 위한 피드백입니다. 각 크루가 아닌 전반적인 진행 상황에 대해 의견을 남겨주세요!

- 사전에 공유한 내용이 잘 전달되었나요?
- 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- 새로운 관점을 배울 수 있었나요?

---
## 질문 리스트

- 자동차미션에서 custom 함수형 인터페이스를 어떻게 사용했는지
    - 표준 함수형 인터페이스로도 가능했을 것 같은데, custom을 선택한 이유?
- 표준 함수형 인터페이스를 사용하는 경우는 언제가 있을지?
- 비즈니스 로직을 담고있을 때 왜 custom 함수형 인터페이스를 써야하는지?
    - 네이밍 이외에 추가적인 장점?
- (Functional Interface 외에)자바 기본 API에 이미 존재하는 형태의 class를 직접 구현해서 이점을 얻는 경우가 어떤 것들이 있을지?
- Functional Interface를 Parameter Type으로 받는 것의 위험성에 대해 고민해보았는지?
    - view에 대한 간접적인 의존
    - 객체의 동작을 외부에서 제어할 수 있게 됨
- 체스 콘솔 게임에서 사용한 것 처럼 사용하면 도메인과 view가 분리되지 않았다는 지적을 받을 수 있을 것
- “Util”이라는 네이밍을 어떨 때 사용하는지?

**Comment by ulimy** [comment]
본인이 경험해보지 못한 부분에 대해 솔직하게 인정하고, 다시 구현하게 된다면 어떤 식으로 할지 말씀해주신 부분이 좋았습니다. 다만 자신이 잘 알고있는 분야에 대해서는 차분히 잘 설명해주셨지만, 자신이 잘 모르는 분야가 나오면 갑자기 말이 조금 빨라진다는 느낌을 받았습니다. 전체적으로 차분하게 잘 이야기 해주셔서 신경쓰일 정도는 아니었어요! 말투가 부드럽고 따뜻해서 좋았습니다. 인터뷰 내내 눈을 맞춰주신 점도 좋았습니다~! 조금 더 자신감 있게 얘기해주셔도 좋을 것 같아요😃

**Comment by tco0427** [comment]
이전에 레벨 말하기 할 때보다는 목소리나 속도가 차분해서 훨씬 듣기 편했던 것 같습니다!
또한 본인의 주관을 마지막에 이야기해주신 점도 좋았고, 저번 레벨 인터뷰 때에는 눈을 마주치는 것에 대한 피드백이 있었던 것으로 기억하는데, 질문을 하고 답변을 할 때 눈을 잘 마주쳐주면서 대화하듯이 답변해주셔서 좋았습니다😁

**Comment by yeon-06** [comment]
시작 전에는 엄청 떠시더니 막상 시작하니까 적당한 템포와 적당한 어조로 잘 말씀하신 것 같아요. 유콩의 여러가지 경험을 들을 수 있어서 좋았습니다! 다만 경험하지 못한 부분에 대해서는 당황해하는게 좀 느껴졌어요. 어떤 식으로 대답해야 자연스레 넘길 수 있을지 고민해보는 것도 좋을 것 같아요😊 (ex: 이 부분에 대해서는 더 학습해오겠습니다 등)

**Comment by YJGwon** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
    - 레벨인터뷰때에 비해서 차분하게 말씀을 잘해주신 것 같아요! 긴장하셨다는데 템포도 적절했고 흔들리는 느낌이 거의 없었어요.
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
    - 정답을 맞추는 것에 대한 불안함이 거의 보이지 않았고 본인의 생각을 자신감있게 잘 말씀해주셨습니다
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
    - 미션을 통해 많은 고민을 해오신 것 같아요. 그리고 오늘 스터디 진행 하면서도 더 많은 생각과 고민을 나누고 싶어하는 의욕을 느꼈어요
- [x] 모르는 부분에 대해 잘 대처했나요?

**Comment by kyukong** [comment]
- [x] 사전에 공유한 내용이 잘 전달되었나요?
- [x] 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- [x] 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- [x] 새로운 관점을 배울 수 있었나요?

많이 긴장했었는데 긴장을 풀어주려고 눈을 맞춰가며 질문해주신 것들이 너무 감사했습니다.ㅠ_ㅠ 새로 알려주신 내용들도 생각해보지 못한 부분이라서 너무 좋았어요!!
